src/d/actor/d_a_npc_cb1.cpp

```diff
@@ -72,532 +88,2993 @@ static dCcD_SrcCyl l_wind_cyl_src = {
+            if(temp >= 0.0f) {
+                setNpcAction(&daNpc_Cb1_c::jumpNpcAction, &temp);
+
+                return TRUE;
+            }
+
+            if(param_1 > 360000.0f) {
+                return FALSE;
+            }
+
+            if(std::fabsf(dComIfGp_getPlayer(0)->current.pos.y - current.pos.y) > 100.0f) {
+                return FALSE;
+            }
+        }
+
+        cXyz temp(current.pos.x, current.pos.y + 100.0f, current.pos.z);
+        cXyz temp2(cM_ssin(*param_2) * 80.0f + temp.x, temp.y, cM_scos(*param_2) * 80.0f + temp.z);
+        routeWallCheck(temp, temp2, param_2);
+    }
+
+    return TRUE;
 }
 
 /* 00005F0C-000062F4       .text searchNpcAction__11daNpc_Cb1_cFPv */
-void daNpc_Cb1_c::searchNpcAction(void*) {
-    /* Nonmatching */
+BOOL daNpc_Cb1_c::searchNpcAction(void*) {
+    f32 temp2;
+
+    if(m8F0 == 0) {
+        cLib_onBit<u32>(attention_info.flags, fopAc_Attn_ACTION_CARRY_e);
+        setAnm(ANM_01);
+    }
+    else if(m8F0 != -1) {
+        s16 temp;
+        if(m8E1) {
+            s16 temp3 = m8E1 < 0 ? -0x4000 : 0x4000;
+            temp = temp3 + cM_atan2s(m91C.x, m91C.z);
+        }
+        else {
+            if(!checkNpcCallCommand()) {
+                temp2 = 0.0f;
+                
+                if(fopAcM_GetSpeedF(this) == 0.0f) {
+                    setNpcAction(&daNpc_Cb1_c::waitNpcAction, NULL);
+
+                    return TRUE;
+                }
+
+                temp2 = 0.0f;
+                temp = current.angle.y;
+            }
+            else {
+                mHasAttention = true;
+
+                fopAc_ac_c* pPlayer = dComIfGp_getPlayer(0);
+
+                BOOL door = pPlayer->eventInfo.checkCommandDoor();
+                f32 dist1 = fopAcM_searchPlayerDistance2(this);
+                f32 dist2 = fopAcM_searchPlayerDistanceXZ2(this);
+
+                if(!door && dist1 < l_HIO.mPlayerChaseDistance * l_HIO.mPlayerChaseDistance) {
+                    temp2 = 0.0f;
+                }
+                else {
+                    temp2 = l_HIO.mChaseDistScale * std::sqrtf(dist2);
+                    temp2 = cLib_maxLimit(temp2, l_HIO.mMaxWalkSpeed);
+                }
+
+                temp = fopAcM_searchPlayerAngleY(this);
+                if(door || !routeCheck(dist2, &temp) || dComIfGp_checkPlayerStatus0(0, daPyStts0_UNK2000000_e | daPyStts0_UNK100_e | daPyStts0_UNK1_e) || ((daPy_py_c*)pPlayer)->checkAttentionLock()) {
+                    temp2 = 0.0f;
+
+                    if(fopAcM_GetSpeedF(this) == 0.0f) {
+                        offNpcCallCommand();
+
+                        setNpcAction(&daNpc_Cb1_c::waitNpcAction, this);
+
+                        return TRUE;
+                    }
+
+                    temp2 = 0.0f;
+                    temp = current.angle.y;
+                }
+
+                if(dist2 < 160000.0f && cLib_distanceAngleS(shape_angle.y, temp) < 0x2000 && std::fabsf(fopAcM_searchPlayerDistanceY(this)) < 100.0f) {
```

Comment:
```suggestion
                if(dist2 < (400.0f * 400.0f) && cLib_distanceAngleS(shape_angle.y, temp) < 0x2000 && std::fabsf(fopAcM_searchPlayerDistanceY(this)) < 100.0f) {
```
Because this is a square dist it can be a bit more clear if the literal is written unsquared